src/Logging.AzureAppServices/src/AzureAppServicesLoggerFactoryExtensions.cs

```diff
@@ -21,15 +21,30 @@ public static class AzureAppServicesLoggerFactoryExtensions
         /// Adds an Azure Web Apps diagnostics logger.
         /// </summary>
         /// <param name="builder">The extension method argument</param>
+        /// <returns></returns>
         public static ILoggingBuilder AddAzureWebAppDiagnostics(this ILoggingBuilder builder)
         {
             var context = WebAppContext.Default;
 
             // Only add the provider if we're in Azure WebApp. That cannot change once the apps started
-            return AddAzureWebAppDiagnostics(builder, context);
+            return AddAzureWebAppDiagnostics(builder, context, null);
         }
 
-        internal static ILoggingBuilder AddAzureWebAppDiagnostics(this ILoggingBuilder builder, IWebAppContext context)
+        /// <summary>
+        /// Adds an Azure Web Apps diagnostics logger. With CustomPrefix the Blob output will be in flat file strucure
+        /// </summary>
+        /// <param name="builder">The extension method argument</param>
+        /// <param name="customPrefix"></param>
+        /// <returns></returns>
+        public static ILoggingBuilder AddAzureWebAppDiagnostics(this ILoggingBuilder builder, string customPrefix)
```

Comment:
I spoke to some folks who were familiar with this component and it seems like changing the layout of the blob file won't be an issue. However, I'd like to suggest a different API.

It seems fairly unintuitive that using this overload would change the folder structure. Even with the API doc in place, this can be a pitfall for users this is not a pattern we have used elsewhere. Instead I propose that we add a configurable option to allow for customized blob name format. Specifically, add a formatter to `AzureBlobLoggerOptions`
```c#
Func<DateTimeOffset, string, string, string> BlobNameFormatter { get; set; }
```

By default this would use the current logic at https://github.com/dotnet/aspnetcore/blob/main/src/Logging.AzureAppServices/src/BlobLoggerProvider.cs#L66 passing in the `_appName`, `_fileName` and a `DateTimeOffset` constructed with the key. To make this more user friendly, we should add an overload that takes a delegate to configure the option in the same style as https://github.com/dotnet/aspnetcore/blob/main/src/Middleware/HttpsPolicy/src/HttpsRedirectionServicesExtensions.cs#L21 so folks can override the default logic

Could you give this a try before I bring this PR to our weekly API review?

Reply:
Hi JunTao, I see what you're saying and totally agree that this is counter intuitive. OK I will look into using delegate like you said. I'd probably start fresh

Reply:
FYI I brought this up in API review and some small changes were suggested, primarily moving the arguments to a context struct so it's more clear what the arguments are for. See the comment at https://github.com/dotnet/aspnetcore/pull/27383#issuecomment-818007583.